GUI/code/GUI_CA_exp_creator.py

```diff
@@ -311,13 +311,34 @@ def write_to_file_CA(ui,CA,CA_window,AP_params):
     """
 
     config_params = get_parameters(CA,'CA',AP_params)
-    config_writer_CA(*config_params)
     data_out_name = config_params[3]
     data_out_path = config_params[4]
+    filename_list = os.listdir(data_out_path)                 # list all the filename in the folder
+    config_writer_CA(*config_params)
     filename = os.path.join(data_out_path, data_out_name + "_CA_config.yml")
-    if warning('The file has been saved. Do you want to exit the window?'):
+    filename_parse = filename.split('/')[-1].strip()          # this contains the filename without the path,e.g. 0_CA_config.yml
```

Comment:
This is ok, but it looks like it will be brittle (easily broken) and difficult to maintain.  I am also seeing a similar code-block copied in two different functions.    It tells me there is almost certainly a better pattern for this.  It also worries me that I am seeing code that should probably be in the model or the controller layer in the gui code.

Reply:
Yes, you are correct.
A similar code-block here is used for different experiments(CA, CV, LSV). There is a certainly neater way to do, which basically combine these functions into one. But different experiments have their own parameters, so customization of different experiments will still be needed in that function. We can definitely do that though. But I think it is safer and easier to understand if we could separate these experiments, even though with some repeated blocks.

Reply:
Thanks for the advice! 
The view right now is largely independent of the model and the controller, but yes, the model and control are not separate. I agree we should fix that, but I'm afraid we don't have enough time before the AMRS. For now, I'm thinking using this for the AMRS and I DO understand it's not as strong, so we can make multiple tests to make sure it won't crash at the ARMS. But in the future, that's definitely something on the schedule. What do you think about this?

Reply:
Raised as issue #129, merging now!

Reply:
:+1: